File: test_client.py
import asyncio
import websockets
import json
import sys

async def test_client(name, room):
    uri = "ws://localhost:8000/ws"
    async with websockets.connect(uri) as websocket:
        print(f"Connected as {name}")
        
        # Login
        login_msg = {
            "type": "Login",
            "payload": { "username": name }
        }
        await websocket.send(json.dumps(login_msg))
        print(f"Sent Login: {login_msg}")
        
        # Join Room
        join_msg = {
            "type": "Join",
            "payload": { "room": room }
        }
        await websocket.send(json.dumps(join_msg))
        print(f"Sent Join: {join_msg}")
        
        # Listen for messages
        while True:
            try:
                message = await asyncio.wait_for(websocket.recv(), timeout=5.0)
                print(f"Received: {message}")
            except asyncio.TimeoutError:
                print("Timeout waiting for message")
                break

if __name__ == "__main__":
    name = sys.argv[1] if len(sys.argv) > 1 else "TestBot"
    room = sys.argv[2] if len(sys.argv) > 2 else "testroom"
    asyncio.run(test_client(name, room))

File: test_client.js
const WebSocket = require('ws');

const name = process.argv[2] || "TestBot";
const room = process.argv[3] || "testroom";
const url = "ws://localhost:8000/ws";

const ws = new WebSocket(url);

ws.on('open', function open() {
    console.log(`Connected as ${name}`);

    const loginMsg = {
        type: "Login",
        payload: { username: name }
    };
    ws.send(JSON.stringify(loginMsg));
    console.log(`Sent Login: ${JSON.stringify(loginMsg)}`);

    const joinMsg = {
        type: "Join",
        payload: { room: room }
    };
    ws.send(JSON.stringify(joinMsg));
    console.log(`Sent Join: ${JSON.stringify(joinMsg)}`);
});

ws.on('message', function incoming(data) {
    console.log(`Received: ${data}`);
});

ws.on('error', function error(err) {
    console.error(`Error: ${err}`);
});

setTimeout(() => {
    console.log("Timeout reached, closing.");
    ws.close();
}, 5000);
